File: Project/dentist-front/src/app/model/appointment.ts
import { Time } from "@angular/common";
import { Dentist } from "./dentist";
import { Patient } from "./patient";
import { Surgery } from "./surgery";

export class Appointment {

    appointmentId:number;
    appointmentDate:Date;
    appointmentTime:Time;
    patientId:number;
    dentistId:number;
    requestId:number;
    branchId:number;
    dentistDTO:Dentist;
    patientDTO:Patient;
    BranchDTO:Surgery;
}

File: Lab6/AdsAppointment/src/main/java/com/ads/model/Appointment.java
package com.ads.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "appointment")
public class Appointment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long appointmentId;
    private LocalDate appointmentDate;
    private LocalTime appointmentTime;

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name="dentist_id", nullable = false)
    private Dentist dentist;

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name="patient_id", nullable = false)
    private Patient patient;

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name="surgery_id", nullable = false)
    private Surgery surgery;
}
